core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/migrate/MigrateElement.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.graph.hook.migrate;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import org.apache.commons.collections.CollectionUtils;
+
+import uk.gov.gchq.gaffer.data.element.function.ElementTransformer;
+import uk.gov.gchq.gaffer.data.element.function.ElementTransformer.Builder;
+import uk.gov.gchq.gaffer.operation.function.migration.Identity;
+import uk.gov.gchq.koryphe.tuple.function.TupleAdaptedFunction;
+
+import java.util.List;
+
+@JsonPropertyOrder(value = {"oldGroup", "newGroup", "toNewFunctions", "toOldFunctions"}, alphabetic = true)
+public class MigrateElement {
+    private String oldGroup;
+    private String newGroup;
+    private ElementTransformer toNewTransform;
+    private ElementTransformer toOldTransform;
+
+    public MigrateElement() {
+    }
+
+    public MigrateElement(final String oldGroup, final String newGroup,
+                          final ElementTransformer toNewTransform,
+                          final ElementTransformer toOldTransform) {
+        this.oldGroup = oldGroup;
+        this.newGroup = newGroup;
+        this.toNewTransform = toNewTransform;
+        this.toOldTransform = toOldTransform;
+        addNewGroupTransform();
+        addOldGroupTransform();
+    }
+
+    public void setOldGroup(final String oldGroup) {
+        this.oldGroup = oldGroup;
+        addOldGroupTransform();
+    }
+
+    public String getOldGroup() {
+        return oldGroup;
+    }
+
+    public void setNewGroup(final String newGroup) {
+        this.newGroup = newGroup;
+        addNewGroupTransform();
```

Comment:
Unfortunately, automatically adding the newGroup transform function in, means if the class is serialised into json it includes this transform. Then if the json is deserialised, it will include the transform twice. Ideally this newGroup (and oldGroup) transform function should only be added at when needed and shouldn't modify the original list of functions.